cli/flox/src/commands/environment.rs

```diff
@@ -357,70 +401,103 @@ impl Activate {
 
         // Add to FLOX_ACTIVE_ENVIRONMENTS so we can detect what environments are active.
         let mut flox_active_environments = activated_environments();
-        flox_active_environments.set_last_active(last_active);
+        if flox_active_environments.is_active(&now_active) {
+            bail!("Environment '{now_active}' is already active");
+        }
+        flox_active_environments.set_last_active(now_active);
 
         // TODO more sophisticated detection?
         let shell = if let Ok(shell) = env::var("SHELL") {
-            shell
+            ShellType::try_from(Path::new(&shell))?
         } else {
             bail!("SHELL must be set");
         };
-        let mut command = Command::new(&shell);
+
+        let prompt_color_1 = env::var("FLOX_PROMPT_COLOR_1")
+            .unwrap_or(utils::colors::LIGHT_BLUE.to_ansi256().to_string());
+        let prompt_color_2 = env::var("FLOX_PROMPT_COLOR_2")
+            .unwrap_or(utils::colors::DARK_PEACH.to_ansi256().to_string());
+
+        // when output is not a tty, and no command is provided
+        // we just print an activation script to stdout
+        //
+        // That script can then be `eval`ed in the current shell,
+        // e.g. in a .bashrc or .zshrc file:
+        //
+        //    eval "$(flox activate)"
+        if !stdout().is_tty() && self.run_args.is_empty() {
```

Comment:
I'd prefer being slightly more specific i.e. by requiring a flag.

Now you get a script when you "probably" want it, i.e. when piping or collecting the output AND not running a command (who's output you likely want).
Also the "only" way to inspect that script you're supposed to `eval` is by piping it into a file or reader `flox activate > ./f; cat f` or `flox activate | less` which might not be obvious to the DAU.

```
flox activate --print-script
flox activate --inplace
flox activate --manual
...
```


Not a big concern it's probably "good enough" but still pinging @ghudgins @jennymahmoudi for consideration

Reply:
I agree that we should be explicit and would prefer a flag for this. Of those options I prefer `--print-script` or just `--print`. Maybe even `--export`?

Reply:
agree that this is not a major concern, but for me to have any opinion I need to understand a bit more about what this is doing. 

If you `flox activate --the-flag-in-question` it will print the example that yannik shared above. Will it also activate the environment or is it NOT activating the environment? (asking bc the `--manual` option makes me think it might not be actually activating). 

If it's still doing the activation, then `--print` feels like the right option.
If it's not doing the activation, then lmk bc that's kind of surprising.

Reply:
`flox activate` normally drops you into an interactive subshell, e.g. bash, with all of our initializations performed:
```
original shell$ flox activate
user in nested shell$ can use programs from environment
user in nested shell$ exit
original shell$ programs have disappeared 
```

This PR adds a version `flox activate` which allows pulling flox initializations into the current shell. This is useful if you don't want to start a subshell, or if you want to add a flox environment to your .bashrc or .zshrc
Don't want to start a subshell:
```
original shell$ eval $(flox activate)
original shell$ can use programs from the environment
original shell$ exit
Terminal closes
```

If I put `eval $(flox activate ~/my-env)` in `.bashrc`, then when I open a Terminal
```
original shell$ can use programs from the environment
```

So to answer your question, technically `eval` is doing the activation in the print case. `flox activate` is printing a script that when `eval`ed will activate your environment.

Reply:
For comparison, `nix` makes it an entirely separate subcommand, `nix print-dev-env`